src/modules/test/main.cpp

```diff
@@ -12,18 +12,43 @@ static std::string str_tolower(std::string s) {
     return s;
 }
 
+std::map<std::string, std::pair<std::shared_ptr<ManagementModule>, std::shared_ptr<MmiSession>>> g_moduleSessionMap;
+
 void RegisterRecipesWithGTest(TestRecipes &testRecipes)
 {
     for (TestRecipe recipe : *testRecipes)
     {
+        ASSERT_STRNE(recipe.m_metadata.m_modulePath.c_str(), "") << "No module path defined!";
+        
+        std::shared_ptr<ManagementModule> module;
+        std::shared_ptr<MmiSession> session;
```

Comment:
Why load *all* modules before running the recipes?

Just load one module at a time when there is a test object for a component that module has, and unload it when the recipe using it is complete.